pymobiledevice3/services/springboard.py

```diff
@@ -17,39 +17,31 @@ class SpringBoardServicesService(LockdownService):
     RSD_SERVICE_NAME = 'com.apple.springboardservices.shim.remote'
     SERVICE_NAME = 'com.apple.springboardservices'
 
-    def __init__(self, lockdown: LockdownServiceProvider):
+    def __init__(self, lockdown: LockdownServiceProvider) -> None:
         if isinstance(lockdown, LockdownClient):
             super().__init__(lockdown, self.SERVICE_NAME)
         else:
             super().__init__(lockdown, self.RSD_SERVICE_NAME)
 
-    def get_icon_state(self, format_version: str = '2'):
+    def get_icon_state(self, format_version: str = '2') -> Mapping:
         cmd = {'command': 'getIconState'}
         if format_version:
             cmd['formatVersion'] = format_version
-
         return self.service.send_recv_plist(cmd)
 
-    def set_icon_state(self, newstate: typing.Mapping = None):
+    def set_icon_state(self, newstate: Optional[Mapping] = None) -> None:
```

Comment:
Actually, this is a list rather than a mapping. get_icon_state also returns list of lists, where each list item corresponds to springboard page. 

In general the output looks like:

```
[[{'bundleIdentifier': 'com.apple.mobilephone',
   'bundleVersion': '36',
   'displayIdentifier': 'com.apple.mobilephone',
   'displayName': 'Phone',
   'iconModDate': datetime.datetime(2023, 9, 2, 6, 59, 21)},
  {'bundleIdentifier': 'com.apple.mobilesafari',
   'bundleVersion': '8615.3.12.10.2',
   'displayIdentifier': 'com.apple.mobilesafari',
   'displayName': 'Safari',
   'iconModDate': datetime.datetime(2023, 9, 2, 6, 59, 21)},
  {'bundleIdentifier': 'com.apple.MobileSMS',
   'bundleVersion': '6000',
   'displayIdentifier': 'com.apple.MobileSMS',
   'displayName': 'Messages',
   'iconModDate': datetime.datetime(2023, 9, 2, 6, 59, 21)},
  {'bundleIdentifier': 'com.apple.Music',
   'bundleVersion': '4022.710.1',
   'displayIdentifier': 'com.apple.Music',
   'displayName': 'Music',
   'iconModDate': datetime.datetime(2023, 9, 9, 19, 24, 2, 316431)}],
 [{'bundleIdentifier': 'com.apple.facetime',
   'bundleVersion': '36',
   'displayIdentifier': 'com.apple.facetime',
   'displayName': 'FaceTime',
   'iconModDate': datetime.datetime(2023, 9, 9, 19, 23, 59, 632491)},
  {'bundleIdentifier': 'com.apple.mobilecal',
   'bundleVersion': '1.0',
   'displayIdentifier': 'com.apple.mobilecal',
   'displayName': 'Calendar',
   'iconModDate': datetime.datetime(2023, 9, 9, 19, 23, 58, 291891)},
  {'bundleIdentifier': 'com.apple.mobileslideshow',
   'bundleVersion': '562.0.100',
   'displayIdentifier': 'com.apple.mobileslideshow',
   'displayName': 'Photos',
   'iconModDate': datetime.datetime(2023, 9, 2, 6, 59, 21)},
  {'bundleIdentifier': 'com.apple.camera',
   'bundleVersion': '3942.0.150',
   'displayIdentifier': 'com.apple.camera',
   'displayName': 'Camera',
   'iconModDate': datetime.datetime(2023, 9, 2, 6, 59, 21)},
  {'bundleIdentifier': 'com.apple.mobilemail',
   'bundleVersion': '3731.700.6',
   'displayIdentifier': 'com.apple.mobilemail',
   'displayName': 'Mail',
   'iconModDate': datetime.datetime(2023, 9, 9, 19, 24, 5, 976007)},
  {'bundleIdentifier': 'com.apple.mobilenotes',
   'bundleVersion': '2255',
   'displayIdentifier': 'com.apple.mobilenotes',
   'displayName': 'Notes',
   'iconModDate': datetime.datetime(2023, 9, 9, 19, 24, 5, 586679)},
  {'bundleIdentifier': 'com.apple.reminders',
   'bundleVersion': '2771',
   'displayIdentifier': 'com.apple.reminders',
   'displayName': 'Reminders',
   'iconModDate': datetime.datetime(2023, 9, 9, 19, 24, 2, 222561)},
  {'bundleIdentifier': 'com.apple.mobiletimer',
   'bundleVersion': '1.0',
   'displayIdentifier': 'com.apple.mobiletimer',
   'displayName': 'Clock',
   'iconModDate': datetime.datetime(2023, 9, 9, 19, 23, 46, 434854)},
  {'bundleIdentifier': 'com.apple.tv',
   'bundleVersion': '1',
   'displayIdentifier': 'com.apple.tv',
   'displayName': 'TV',
   'iconModDate': datetime.datetime(2023, 9, 9, 19, 24, 5, 718878)},
  {'bundleIdentifier': 'com.apple.podcasts',
   'bundleVersion': '4022.700.8',
   'displayIdentifier': 'com.apple.podcasts',
   'displayName': 'Podcasts',
   'iconModDate': datetime.datetime(2023, 9, 9, 19, 23, 56, 514462)},
  {'bundleIdentifier': 'com.apple.AppStore',
   'bundleVersion': '1',
   'displayIdentifier': 'com.apple.AppStore',
   'displayName': 'App Store',
   'iconModDate': datetime.datetime(2023, 9, 2, 6, 59, 21)},
  {'bundleIdentifier': 'com.apple.Maps',
   'bundleVersion': '2751.36.4.20.6',
   'displayIdentifier': 'com.apple.Maps',
   'displayName': 'Maps',
   'iconModDate': datetime.datetime(2023, 9, 9, 19, 23, 59, 174699)},
  {'bundleIdentifier': 'com.apple.Health',
   'bundleVersion': '3232.6.1.8',
   'displayIdentifier': 'com.apple.Health',
   'displayName': 'Health',
   'iconModDate': datetime.datetime(2023, 9, 9, 19, 23, 43, 746924)},
  {'bundleIdentifier': 'com.apple.Passbook',
   'bundleVersion': '1505.26.1.1',
   'displayIdentifier': 'com.apple.Passbook',
   'displayName': 'Wallet',
   'iconModDate': datetime.datetime(2023, 9, 9, 19, 23, 47, 648443)},
  {'bundleIdentifier': 'com.apple.Preferences',
   'bundleVersion': '1173.3.17',
   'displayIdentifier': 'com.apple.Preferences',
   'displayName': 'Settings',
   'iconModDate': datetime.datetime(2023, 9, 2, 6, 59, 21)}],
 [{'bundleIdentifier': 'com.apple.weather',
   'bundleVersion': '358',
   'displayIdentifier': 'com.apple.weather',
   'displayName': 'Weather',
   'iconModDate': datetime.datetime(2023, 9, 9, 19, 23, 54, 546409)},
  {'bundleIdentifier': 'com.apple.stocks',
   'bundleVersion': '1196.4',
   'displayIdentifier': 'com.apple.stocks',
   'displayName': 'Stocks',
   'iconModDate': datetime.datetime(2023, 9, 9, 19, 24, 3, 917617)},
  {'bundleIdentifier': 'com.apple.findmy',
   'bundleVersion': '310.6.1',
   'displayIdentifier': 'com.apple.findmy',
   'displayName': 'Find My',
   'iconModDate': datetime.datetime(2023, 9, 9, 19, 23, 46, 311491)},
  {'bundleIdentifier': 'com.apple.Home',
   'bundleVersion': '733.7.11',
   'displayIdentifier': 'com.apple.Home',
   'displayName': 'Home',
   'iconModDate': datetime.datetime(2023, 9, 9, 19, 23, 59, 809049)},
  {'bundleIdentifier': 'com.apple.iBooks',
   'bundleVersion': '5800.52',
   'displayIdentifier': 'com.apple.iBooks',
   'displayName': 'Books',
   'iconModDate': datetime.datetime(2023, 9, 9, 19, 23, 59, 880238)},
  {'bundleIdentifier': 'com.apple.MobileStore',
   'bundleVersion': '1',
   'displayIdentifier': 'com.apple.MobileStore',
   'displayName': 'iTunes Store',
   'iconModDate': datetime.datetime(2023, 9, 9, 19, 23, 57, 503829)},
  {'bundleIdentifier': 'com.apple.Fitness',
   'bundleVersion': '1.0',
   'displayIdentifier': 'com.apple.Fitness',
   'displayName': 'Fitness',
   'iconModDate': datetime.datetime(2023, 9, 9, 19, 24, 24, 148711)},
  {'bundleIdentifier': 'com.apple.Bridge',
   'bundleVersion': '1.0',
   'displayIdentifier': 'com.apple.Bridge',
   'displayName': 'Watch',
   'iconModDate': datetime.datetime(2023, 9, 9, 19, 23, 58, 796589)},
  {'bundleIdentifier': 'com.apple.MobileAddressBook',
   'bundleVersion': '1.0',
   'displayIdentifier': 'com.apple.MobileAddressBook',
   'displayName': 'Contacts',
   'iconModDate': datetime.datetime(2023, 9, 9, 19, 23, 54, 262279)},
  {'bundleIdentifier': 'com.apple.Translate',
   'bundleVersion': '218.7',
   'displayIdentifier': 'com.apple.Translate',
   'displayName': 'Translate',
   'iconModDate': datetime.datetime(2023, 9, 9, 19, 23, 56, 332764)},
  {'bundleIdentifier': 'com.apple.DocumentsApp',
   'bundleVersion': '280.8',
   'displayIdentifier': 'com.apple.DocumentsApp',
   'displayName': 'Files',
   'iconModDate': datetime.datetime(2023, 9, 9, 19, 23, 54, 848550)},
  {'bundleIdentifier': 'com.apple.shortcuts',
   'bundleVersion': '1505.3',
   'displayIdentifier': 'com.apple.shortcuts',
   'displayName': 'Shortcuts',
   'iconModDate': datetime.datetime(2023, 9, 9, 19, 24, 5, 751963)},
  {'displayName': 'Utilities',
   'iconLists': [[{'bundleIdentifier': 'com.apple.VoiceMemos',
                   'bundleVersion': '2.3',
                   'displayIdentifier': 'com.apple.VoiceMemos',
                   'displayName': 'Voice Memos',
                   'iconModDate': datetime.datetime(2023, 9, 9, 19, 23, 54, 994632)},
                  {'bundleIdentifier': 'com.apple.compass',
                   'bundleVersion': '1.0',
                   'displayIdentifier': 'com.apple.compass',
                   'displayName': 'Compass',
                   'iconModDate': datetime.datetime(2023, 9, 9, 19, 24, 4, 111844)},
                  {'bundleIdentifier': 'com.apple.measure',
                   'bundleVersion': '162.4.1',
                   'displayIdentifier': 'com.apple.measure',
                   'displayName': 'Measure',
                   'iconModDate': datetime.datetime(2023, 9, 9, 19, 24, 0, 738454)},
                  {'bundleIdentifier': 'com.apple.Magnifier',
                   'bundleVersion': '1',
                   'displayIdentifier': 'com.apple.Magnifier',
                   'displayName': 'Magnifier',
                   'iconModDate': datetime.datetime(2023, 9, 9, 19, 23, 56, 799478)},
                  {'bundleIdentifier': 'com.apple.calculator',
                   'bundleVersion': '1.0',
                   'displayIdentifier': 'com.apple.calculator',
                   'displayName': 'Calculator',
                   'iconModDate': datetime.datetime(2023, 9, 9, 19, 23, 54, 227252)}]],
   'listType': 'folder'},
  {'bundleIdentifier': 'com.apple.freeform',
   'bundleVersion': '138',
   'displayIdentifier': 'com.apple.freeform',
   'displayName': 'Freeform',
   'iconModDate': datetime.datetime(2023, 9, 9, 19, 23, 43, 294307)},
  {'bundleIdentifier': 'com.apple.tips',
   'bundleVersion': '9999.99.99',
   'displayIdentifier': 'com.apple.tips',
   'displayName': 'Tips',
   'iconModDate': datetime.datetime(2023, 9, 9, 19, 23, 55, 760963)},
]]
```

Reply:
thanks! i'll add a test and fix that aswell